hw/top_earlgrey/syn/chip_earlgrey_asic.sdc

```diff
@@ -50,35 +50,71 @@ set USB_CLK_PIN u_ast/clk_src_usb_o
+                                 -to   [get_pins -hierarchical -filter "full_name =~ *u_dio_pad*/*/A"]
+
+  # dn output timing
+  # note that there is a path to the OE as well due to virtual open drain emulation in the pad wrapper (although it is likely not being used for USB).
+  set_max_delay ${MAX_USB_DELAY} -from [get_pins top_earlgrey/u_usbdev/usbdev_impl/u_usb_fs_nb_pe/u_usb_fs_tx/u_usb_dn_o_flop/${FLOP_PATH}/Q] \
+                                 -to   [get_pins -hierarchical -filter "full_name =~ *u_dio_pad*/*/OE"]
+  set_max_delay ${MAX_USB_DELAY} -from [get_pins top_earlgrey/u_usbdev/usbdev_impl/u_usb_fs_nb_pe/u_usb_fs_tx/u_usb_dn_o_flop/${FLOP_PATH}/Q] \
+                                 -to   [get_pins -hierarchical -filter "full_name =~ *u_dio_pad*/*/A"]
+
+  # We reuse the same set_max_delay constraints as for the driver paths to stay on the safe side
+  # (there is more skew budget on the receiver side according to the spec, but we shouldn't be using that 
+  # up since there would otherwise be no margin for cable and PCB skew).
+  #
+  # The USBDEV has both a regular and a differential amplifier input mode.
+  # For the former, the skew only matters up to the differential amplifier inputs.
+  # For the latter, we need to constrain the skew up to the flop inputs.
+
+  # dp input timing to differential receiver
+  set_max_delay ${MAX_USB_DELAY} -from [get_ports USB_P]                                        \
+                                 -to [get_pins -leaf -filter {@pin_direction == in} -of_objects \
+                                        [get_nets -segments -of_objects                         \
+                                          [get_pins u_prim_usb_diff_rx/input_pi]]]
+
+  # dn input timing to differential receiver
+  set_max_delay ${MAX_USB_DELAY} -from [get_ports USB_N]                                        \
+                                 -to [get_pins -leaf -filter {@pin_direction == in} -of_objects \
+                                        [get_nets -segments -of_objects                         \
+                                          [get_pins u_prim_usb_diff_rx/input_ni]]]
+
+  # dp/dn input timing to regular regs
+  set_max_delay ${MAX_USB_DELAY} -from [get_pins -hierarchical -filter "full_name =~ *u_dio_pad*/*/Y"] \
+                                 -to   [get_pins top_earlgrey/u_usbdev/i_usbdev_iomux/cdc_io_to_usb/u_sync_1/${FLOP_PATH}/D]
 }
 
-set_max_delay 5 -from [get_ports USB_N] -to [get_pins ${USBDEV_IOMUX_PATH}/${USB_N_PIN}]
-set_max_delay 5 -from [get_ports USB_P] -to [get_pins ${USBDEV_IOMUX_PATH}/${USB_P_PIN}]
-set_max_delay 5 -from [get_ports USB_*] -to [get_pins ${USBDEV_IOMUX_PATH}/${USB_PIN}]
-set_max_delay 5 -from [get_pins ${USBDEV_OUTREG_PATH}/${USB_D_PIN}] -to [get_ports USB_*]
-set_max_delay 5 -from [get_pins ${USBDEV_OUTREG_PATH}/${USB_OE_PIN}] -to [get_ports USB_*]
```

Comment:
@arnonsha it turned out that the paths were not fully up to date. This updates both the generic 5ns constraints, and adds a few more to make sure the skew is minimized. It is probably also be a good idea to create a placement guide to make sure the `u_usb_fs_tx` and `u_prim_usb_diff_rx` and `i_usbdev_iomux` submodules are placed as closely to the USB pads as possible.